docs/proposals/ose-rebase.md

```diff
@@ -0,0 +1,39 @@
+## Rebase OSE on Origin
+
+Today, code in OSE lands by fetching and merging Origin manually. OSE-specific changes
+are scattered in the repo and it's hard to reason about any conflicts. This doc proposes
+moving all the changes in OSE that diverge from Origin into a set of commits that
+will sit on top of the Origin code[0] and a Jenkins job that will be rebasing master and
+all active release branches on a daily basis.
+
+There are two types of commits that need to be carried on top of Origin: 
+1. long-term carries related to branding and release tooling (master and release branches)
+2. CVEs (release branches only)
+
+Rebases are going to be simpler for the master branch. The OSE master will be hard-reset to
+the Origin master and any carry commits will be applied on top. If any conflicts are hit, manual
+resolution will be required. In the future, we can extend the conflict resolution mechanism in
+Jenkins to send an e-mail either to the author and committer of the carry commit or we can store
+e-mails in the commit description and use those instead.
+
+Rebases for release branches are less trivial because they may contain backported fixes for CVEs
+that do not exist in the respective Origin branches. In such a case, backports should continue
+to be carried until we identify that the backport exists in the Origin branch and drop the OSE commit.
+How? Assuming the backport ends up w/o any additional change in Origin then cherry-picking
+the OSE commit will produce no additional commit. Conflicts in backports need to be resolved manually.
+
+Code that already exists in Origin master and needs to be backported in an enterprise branch can be simply
+backported in the respective Origin release branch and the Jenkins job should be responsible for merging
+the Origin release branch into the OSE release branch.
+
+Naming conventions we will have to enforce in commit messages in OSE:
+* commits that need to be carried long-term such as branding or release tooling changes will start with [CARRY]
+* commits for CVEs may contain a [CVE] label. These commits should be the only fixes that developers
+would need to open pull requests for in the OSE repo (apart from rebases, assuming an automated rebase
+fails, and it needs to run manually due to conflicts).
```

Comment:
Some of the diffs look like simple `sed 's/Origin/Container Platform/g'` type diffs -- do we want to be applying git patches or do we want to be running diff scripts?